app/code/Magento/Ui/view/frontend/web/js/view/messages.js

```diff
@@ -68,8 +70,8 @@ define([
             // Hide message block if needed
             if (isHidden) {
                 setTimeout(function () {
-                    $(self.selector).hide('blind', {}, 500);
-                }, 5000);
+                    $(self.selector).hide('blind', {}, self.hideSpeed);
```

Comment:
Can you please .bind context to timeout callback function and remove `self` usage?

Reply:
Hey Vladimir,

Like this?

```
        onHiddenChange: function (isHidden) {
            // Hide message block if needed
            if (isHidden) {
                setTimeout(function () {
                    $(this.selector).hide('blind', {}, this.hideSpeed);
                }.bind(this), this.hideTimeout);
            }
        }
```